### PR TITLE
Added RepondDecorators to unmarshall primitive types

### PR DESCRIPTION
**Pull request by brendandixon**
Added RepondDecorators to unmarshall primitive types
Corrected invocation of Prepare inspector / authorizer

Signed-off-by: Brendan Dixon <brendand@microsoft.com>

### CODE REVIEW

File: autorest/responder.go [hunk 1]
```diff
@@ -97,6 +97,90 @@ func ByClosingIfError() RespondDecorator {
 	}
 }
 
+// ByUnmarshallingBool returns a RespondDecorator that decodes the http.Response Body into a bool
+// pointed to by b.
+func ByUnmarshallingBool(b *bool) RespondDecorator {
```
Comment (ahmetalpbalkan) [inline autorest/responder.go]: what's the use case for these responders?

File: autorest/mocks/helpers.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ const (
 	TestAuthorizationHeader = "BEARER SECRETTOKEN"
 
 	// TestBadURL is a malformed URL
-	TestBadURL = ""
+	TestBadURL = "                               "
```
Comment (ahmetalpbalkan) [inline autorest/mocks/helpers.go]: `strings.Repeat`

File: autorest/responder.go (hunk 1 above)
Comment (brendandixon) [inline autorest/responder.go]: Swagger expressly allows for primitive return types. Go's JSON decoder does not correctly handle those cases. We need to explicitly catch and convert using these responders.

File: autorest/mocks/helpers.go (hunk 2 above)
Comment (brendandixon) [inline autorest/mocks/helpers.go]: Whatever. Blech. Ok.

Comment (brendandixon) [inline autorest/mocks/helpers.go]: Actually, no can do. TestBadURL is a constant (and should be). Go disallows constant initialization from code (that is, the call to strings.Repeat is disallowed).

File: autorest/responder.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline autorest/responder.go]: Ah ok.

File: autorest/mocks/helpers.go (hunk 2 above)
Comment (ahmetalpbalkan) [inline autorest/mocks/helpers.go]: that's fine then.

**Pull request merged**